src/util/types.ts

```diff
@@ -42,10 +42,31 @@ export interface RelayerState {
     readonly selectedToken: Token | null;
 }
 
+export enum StepKind {
+    BuySellLimit,
+}
+
+export interface StepBuySellLimitOrder {
+    kind: StepKind.BuySellLimit;
+    amount: BigNumber;
+    price: BigNumber;
+    side: OrderSide;
+}
+
+export type Step = StepBuySellLimitOrder;
+
+export interface StepsModalState {
+    readonly isVisible: boolean;
```

Comment:
What do you think about doing this a computed property through a selector? Maybe the modal is visible when the current step is not null?